### PR TITLE
KEYCLOAK-13085 pt_BR messages for login and email

### PR DESCRIPTION
**Pull request by helder-jaspion**

### CODE REVIEW

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 1]
```diff
@@ -2,42 +2,51 @@ doLogIn=Entrar
...
+doBack=Voltar
 doYes=Sim
 doNo=N\u00E3o
-doContinue=Continue
+doContinue=Continuar
+doIgnore=Ignorar
 doAccept=Aceitar
 doDecline=Rejeitar
-doForgotPassword=Esqueceu sua senha?
+doForgotPassword=Esqueceu a senha?
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Do you really want to change this? If you don't mind, I would just leave as is.

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 2]
```diff
@@ -2,42 +2,51 @@ doLogIn=Entrar
...
 doClickHere=Clique aqui
 doImpersonate=Personificar
+doTryAgain=Tente novamente
+doTryAnotherWay=Tente Outra Maneira
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: By following the same pattern to other messages, I think we should have `Tente outra maneira`. Talvez, `Tente outra forma` seja mais correto ?

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 3]
```diff
@@ -2,42 +2,51 @@ doLogIn=Entrar
...
-registerWithTitle=Registre-se com {0}
-registerWithTitleHtml=Registre-se com <strong>{0}</strong>
+registerTitle=Registre-se
 loginTitle=Entrar em {0}
-loginTitleHtml=Entrar em <strong>{0}</strong>
-impersonateTitle={0} Impersonate User
-impersonateTitleHtml=<strong>{0}</strong> Impersonate User
-realmChoice=Realm
+loginTitleHtml={0}
+impersonateTitle={0} Personificar Usu\u00E1rio
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: `Impersonar` seems more correct to me.

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 4]
```diff
@@ -2,42 +2,51 @@ doLogIn=Entrar
...
+impersonateTitleHtml=<strong>{0}</strong> Personificar Usu\u00E1rio
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Same as above.

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 5]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
 emailVerified=E-mail verificado
-gssDelegationCredential=gss delega\u00E7\u00E3o credencial
+gssDelegationCredential=Delega\u00E7\u00E3o de Credenciais GSS
+
+profileScopeConsentText=Perfil de usu\u00E1rio
+emailScopeConsentText=Endere\u00E7o de e-mail
+addressScopeConsentText=Endere\u00E7o
+phoneScopeConsentText=N\u00FAmero de telefone
+offlineAccessScopeConsentText=Acesso Offline
+samlRoleListScopeConsentText=Minhas Fun\u00E7\u00F5es
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: I don't have a better translation for roles :) But `funções` sounds odd. Maybe `Meus Perfis de Acesso` ?

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 6]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
+rolesScopeConsentText=Fun\u00E7\u00F5es do usu\u00E1rio
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Maybe `Perfis de Acesso do Usuário`. Too big though ...

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 7]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
+
+restartLoginTooltip=Reiniciar o login
+
+loginTotpIntro=Voc\u00EA precisa configurar um gerador de senha de uso \u00FAnico para acessar esta conta
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: I think `gerador de código de uso único` makes more sense for us.

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 8]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
+loginTotpStep1=Instale um dos seguintes aplicativos no seu celular:
+loginTotpStep2=Abra o aplicativo e digitalize o c\u00F3digo de barras:
+loginTotpStep3=Digite o c\u00F3digo \u00FAnico fornecido pelo aplicativo e clique em Ok para concluir a configura\u00E7\u00E3o.
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Here you are using `código único` so my comment to `loginTotpIntro` makes sense?

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 9]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
+loginTotpStep3DeviceName=Forne\u00E7a um nome de dispositivo para ajud\u00E1-lo a gerenciar seus dispositivos OTP.
+loginTotpManualStep2=Abra o aplicativo e digite a chave:
+loginTotpManualStep3=Use os seguintes valores de configura\u00E7\u00E3o se o aplicativo permitir defini-los:
+loginTotpUnableToScan=N\u00E3o foi poss\u00EDvel digitalizar?
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Maybe `Não foi possível ler o QRCode?` ?

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 10]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
+loginTotpScanBarcode=Digitalizar c\u00F3digo de barras?
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: `Ler QRCode?`. My point is that I hear more `ler` than `digitalizar` with respect to qrcodes.

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 11]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
+emailLinkIdp4=Se voc\u00EA j\u00E1 verificou o email em outro navegador
+emailLinkIdp5=para continuar.
 
-backToLogin=&laquo; Voltar
+backToLogin=&laquo; Voltar ao Login
 
 emailInstruction=Digite seu nome de usu\u00E1rio ou endere\u00E7o de e-mail e n\u00F3s lhe enviaremos instru\u00E7\u00F5es sobre como criar uma nova senha.
 
-copyCodeInstruction=Por favor, copie o c\u00F3digo e cole-o em seu aplicativo:
+copyCodeInstruction=Por favor, copie este c\u00F3digo e cole-o em seu aplicativo:
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Do you really want to change this?

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 12]
```diff
@@ -65,73 +74,119 @@ region=Estado
...
 missingLastNameMessage=Por favor, informe o sobrenome.
 missingEmailMessage=Por favor, informe o e-mail.
 missingUsernameMessage=Por favor, informe o nome de usu\u00E1rio.
 missingPasswordMessage=Por favor, informe a senha.
 missingTotpMessage=Por favor, informe o c\u00F3digo autenticador.
+missingTotpDeviceNameMessage=Por favor, informe o nome do dispositivo.
 notMatchPasswordMessage=As senhas n\u00E3o coincidem.
 
 invalidPasswordExistingMessage=Senha atual inv\u00E1lida.
+invalidPasswordBlacklistedMessage=Senha inv\u00E1lida: a senha est\u00E1 na lista negra.
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Maybe `Senha inválida de acordo com as políticas estabelecidas para o domínio` ?

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 13]
```diff
@@ -202,13 +264,108 @@ invalidCodeMessage=C\u00F3digo inv\u00E1lido, por favor fa\u00E7a login novament
...
-realmSupportsNoCredentialsMessage=O realm n\u00E3o suporta qualquer tipo de credencial.
-identityProviderNotUniqueMessage=O realm suporta m\u00FAltiplos provedores de identidade. N\u00E3o foi poss\u00EDvel determinar qual o provedor de identidade deve ser usado para se autenticar.
+staleCodeMessage=Esta p\u00E1gina n\u00E3o \u00E9 mais v\u00E1lida. Volte ao seu aplicativo e fa\u00E7a login novamente
+realmSupportsNoCredentialsMessage=O dom\u00EDnio n\u00E3o suporta qualquer tipo de credencial.
+credentialSetupRequired=N\u00E3o \u00E9 poss\u00EDvel fazer o login, \u00E9 necess\u00E1ria configura\u00E7\u00E3o de credencial.
+identityProviderNotUniqueMessage=O dom\u00EDnio suporta m\u00FAltiplos provedores de identidade. N\u00E3o foi poss\u00EDvel determinar qual o provedor de identidade deve ser usado para se autenticar.
 emailVerifiedMessage=O seu endere\u00E7o de e-mail foi confirmado.
+staleEmailVerificationLink=O link em que voc\u00EA clicou \u00E9 um link antigo e n\u00E3o \u00E9 mais v\u00E1lido. Talvez voc\u00EA j\u00E1 tenha verificado seu e-mail.
+identityProviderAlreadyLinkedMessage=A identidade federada retornada por {0} j\u00E1 est\u00E1 vinculada a outro usu\u00E1rio.
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: To technical? Maybe`A conta retornada do {0} já está vinculada ...` ?

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 14]
```diff
@@ -202,13 +264,108 @@ invalidCodeMessage=C\u00F3digo inv\u00E1lido, por favor fa\u00E7a login novament
...
 backToApplication=&laquo; Voltar para o aplicativo
-missingParameterMessage=Par\u00E2metros que faltam\: {0}
+missingParameterMessage=Par\u00E2metros ausentes\: {0}
 clientNotFoundMessage=Cliente n\u00E3o encontrado.
+clientDisabledMessage=Cliente desativado.
 invalidParameterMessage=Par\u00E2mentro inv\u00E1lido\: {0}
-alreadyLoggedIn=You are already logged in.
+alreadyLoggedIn=Voc\u00EA j\u00E1 est\u00E1 logado.
+differentUserAuthenticated=Voc\u00EA j\u00E1 est\u00E1 autenticado como usu\u00E1rio diferente ''{0}'' nesta sess\u00E3o. Por favor, saia primeiro.
+brokerLinkingSessionExpired=Vincula\u00E7\u00E3o de conta do broker solicitada, mas a sess\u00E3o atual n\u00E3o \u00E9 mais v\u00E1lida.
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: `broker` == `provedor de identidade`.

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 15]
```diff
@@ -202,13 +264,108 @@ invalidCodeMessage=C\u00F3digo inv\u00E1lido, por favor fa\u00E7a login novament
...
+proceedWithAction=&raquo; Clique aqui para continuar
+
+requiredAction.CONFIGURE_TOTP=Configurar OTP
+requiredAction.terms_and_conditions=Termos e Condi\u00E7\u00F5es
+requiredAction.UPDATE_PASSWORD=Atualizar Senha
+requiredAction.UPDATE_PROFILE=Atualizar Perfil
+requiredAction.VERIFY_EMAIL=Verificar E-mail
+
+doX509Login=Voc\u00EA ser\u00E1 logado como\:
+clientCertificate=Certificado de cliente X509\:
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: `Certificado X509 do cliente`

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties [hunk 16]
```diff
@@ -202,13 +264,108 @@ invalidCodeMessage=C\u00F3digo inv\u00E1lido, por favor fa\u00E7a login novament
...
+console-confirm-password=Confirma\u00E7\u00E3o de senha:
+console-update-password=\u00C9 necess\u00E1rio atualizar sua senha.
+console-verify-email=Voc\u00EA precisa verificar seu endere\u00E7o de e-mail.  Enviamos um email para {0} que cont\u00E9m um c\u00F3digo de verifica\u00E7\u00E3o.  Digite este c\u00F3digo no campo abaixo.
+console-email-code=C\u00F3digo do email:
+console-accept-terms=Aceita os termos? [s/n]:
+console-accept=s
+
+# Openshift messages
+openshift.scope.user_info=Informa\u00E7\u00E3o do usu\u00E1rio
+openshift.scope.user_check-access=Informa\u00E7\u00E3o do acesso do usu\u00E1rio
```
Comment (pedroigor) [inline themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: I think `Informações de acesso do usuário` is better.

**Review by pedroigor — CHANGES_REQUESTED** [review]

**Comment by pedroigor** [comment]
@helder-jaspion Thanks for this one. Great work!

Just some minor comments. Please, let me know if you think they make sense.

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 3 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: `Impersonar` is an anglicism. The correct way would be 'Personificar'.
https://ciberduvidas.iscte-iul.pt/consultorio/perguntas/o-anglicismo-impersonar-e-imitar/22977

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 8 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Yes, it surely does

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 1 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Actually I don't mind, I'm leaving as is

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 2 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Ok

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 5 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Yes, much better

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 6 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Changed as suggested

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 7 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Ok

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 10 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: `QRCode` really sounds better than `Código de barras`. thanks

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 11 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Actually I don't. thanks

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 12 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Much better

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 14 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Ok. thanks

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 13 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Ok

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 15 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Ok, thanks

File: themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties (hunk 16 above)
Comment (helder-jaspion) [reply themes/src/main/resources-community/theme/base/login/messages/messages_pt_BR.properties]: Yes, it is. thanks

**Review by pedroigor — APPROVED** [review]
Thanks.

**Pull request merged**